pkg/rbac/userData.go

```diff
@@ -340,15 +342,18 @@ func (user *UserDataCache) getSSRRforNamespace(ctx context.Context, cache *Cache
 func (user *UserDataCache) getNamespacedResources(cache *Cache, ctx context.Context,
 	clientToken string) (*UserDataCache, error) {
 	defer metric.SlowLog("UserDataCache::getNamespacedResources", 250*time.Millisecond)()
-	// check if we already have user's namespaced resources in userData cache and check if time is expired
+
+	// Lock the cache
 	user.nsrCache.lock.Lock()
 	defer user.nsrCache.lock.Unlock()
+	user.clustersCache.lock.Lock() // Lock the ManagedCluster cache because this function will update it.
+	defer user.clustersCache.lock.Unlock()
```

Comment:
not sure we need these locks for clusterCache here

Reply:
+1

Reply:
FYI @jlpadilla I pushed the change to remove it. Will add it back if needed.

Reply:
Thanks for debugging and fixing!